### PR TITLE
Deal with file closing

### PR DESCRIPTION
**Pull request by alessiosavi**
Close the file when necessary
Use `%d` during print instead of `%u` due to signed int

### CODE REVIEW

File: src/agoo.c [hunk 1]
```diff
@@ -233,20 +233,27 @@ agoo_load_graphql(agooErr err, const char *path, const char *filename) {
...
     }
     if (0 > (size = ftell(f))) {
+      fclose(f);
 	return agoo_err_no(err, "Failed to ftell %s.", filename);
     }
     rewind(f);
     if (0 < size) {
 	if (NULL == (sdl = AGOO_MALLOC(size))) {
+    fclose(f);
+    free(sdl);
```
Comment (ohler55) [inline src/agoo.c]: No need for `free()` here as the allocation failed.

**Review by ohler55 — CHANGES_REQUESTED** [review]
It does make sense to close the file if fseek or ftell fail. I had assumed a error on those operation would mean the file had been closed but you are correct, it might still be open.

Good catch on the `%u`.

Just remove the `free(sdl)` from that one spot and I'll approve and merge.

File: src/agoo.c (hunk 1 above)
Comment (alessiosavi) [reply src/agoo.c]: Hi @ohler55, thank you for the time spent!

Fix in 06f1201c21faab47c5f9d1e13611a778be666131

**Comment by ohler55** [comment]
Thanks for the PR.

**Comment by alessiosavi** [comment]
You are welcome, thank you for your free software (:

**Pull request merged**